assemble/bin/accumulo-cluster

```diff
@@ -43,52 +43,74 @@ function invalid_args {
   exit 1
 }
 
-function verify_config {
+function parse_config {
   if [[ -f ${conf}/slaves ]]; then
     echo "ERROR: A 'slaves' file was found in ${conf}/"
-    echo "Accumulo now reads tablet server hosts from 'tservers' and requires that the 'slaves' file not be present to reduce confusion."
+    echo "Accumulo now uses cluster host configuration information from 'cluster.yml' and requires that the 'slaves' file not be present to reduce confusion."
     echo "Please rename the 'slaves' file to 'tservers' or remove it if both exist."
     exit 1
   fi
 
-  if [[ ! -f ${conf}/tservers ]]; then
-    echo "ERROR: A 'tservers' file was not found at ${conf}/tservers"
-    echo "Please make sure it exists and is configured with tablet server hosts."
+  if [[ ! -f ${conf}/cluster.yml ]]; then
+    echo "ERROR: A 'cluster.yml' file was not found at ${conf}/cluster.yml"
+    echo "Please make sure it exists and is configured with the host information. Run 'create-config' to create an example configuration."
     exit 1
   fi
 
-  unset manager1
-  if [[ -f "${conf}/$manager_file" ]]; then
-    manager1=$(grep -Ev '(^#|^\s*$)' "${conf}/$manager_file" | head -1)
+  config=$(${accumulo_cmd} org.apache.accumulo.core.conf.cluster.ClusterConfigParser ${conf}/cluster.yml)
+
+  declare -A HOSTS
+
+  HOSTS["MANAGERS"]=$(grep managers $config | awk -F':' '{ print $2 }')
+  HOSTS["MONITORS"]=$(grep monitors $config | awk -F':' '{ print $2 }')
+  HOSTS["GC"]=$(grep gc $config | awk -F':' '{ print $2 }')
+  HOSTS["TRACER"]=$(grep tracer $config | awk -F':' '{ print $2 }')
+  HOSTS["TSERVERS"]=$(grep tservers $config | awk -F':' '{ print $2 }')
```

Comment:
Don't rely on bash 4.0 associative arrays. They won't work on a Mac. It's also unnecessary, because you can just as easily do `MANAGER_HOSTS=...` or similar. You don't actually need all hosts to be stored in the same variable. I'm also a bit confused by the grepping over a variable instead of a file handle. I don't think that will work.

If the output was something like:

```
manager:host1,host2,host3
gc:host4
```

Then `$config` would look like (a space separated array):
```
manager:host1,host2,host3 gc:host4
```

then you could either loop through the array until you found an item starting with `manager:`, or just grep for the manager chunk at word boundaries:

```bash
MANAGER_HOSTS=$(echo "$config" | grep '\<managers:[^ ]*' | cut -f2- -d: | tr ',' ' ')
```

Also be careful about using colon as a delimiter for hosts, because hosts could be specified as IPv6, or we may wish to add port numbers at some point. In my example above, it's okay, because it only removes the first colon with the trailing dash in the field specifier, `-f2-` (interpret as fields 2 and later; the delimiter was specified as a colon with `-d:`).